tests/integration/detection/test_model.py

```diff
@@ -45,6 +45,11 @@ def test__init() -> None:
     assert model == Model(name)
 
 
+def test__init__validate_name() -> None:
+    with pytest.raises(ValueError):
+        Model("")
+
```

Comment:
For this and other definitions, we should be testing the `create` method as well

Reply:
Actually, should the be removed entirely in favor of the unit tests?

Reply:
I thought I had removed this, will remove integration tests in favor of unit tests.

Reply:
Done, some tests for workflow are kept in integration (test case init) as they require define_worfklow to be called (<TestCase> must be subclassed).